go/arrow/encoded/ree_utils.go

```diff
@@ -23,38 +23,55 @@ import (
 	"github.com/apache/arrow/go/v12/arrow"
 )
 
-// FindPhysicalOffset performs a binary search on the run-ends to return
+// FindPhysicalIndex performs a binary search on the run-ends to return
```

Comment:
Oh, I should do this rename on the C++ impl as well, right? From `FindPhysicalOffset(offset)` to `PhysicalIndex(index)` and have `FindPhysicalOffset()` defined in terms of `FindPhysicalIndex(offset)`?

Reply:
It would make sense to do so. Probably a good idea.

Reply:
Just did it. Makes things easier to understand.